### DISCUSSION TITLE
auth session runs out no response

### DISCUSSION THREAD

**Discussion by adeiming** [livewire/livewire#3125] [7. Bugs (archived)]
I opened 2 browser tabs, each was logged in, on that page there was a list of users. When the first tab logs out, and opens tab 2 and then deletes data, no message comes out, even though in XHR the status code is 302 then redirect to the login route.

I hope at least there is a message that the user has logged out or run out of sessions.

- Livewire version: [2.x]
- Laravel version: [8.x]
- Browser: [Chrome]

**Comment by joshhanley** [comment]
@adeiming can you please be specific about what patch versions of Livewire and Laravel you are using? Both have had a lot of changes over the last couple of months.

You will probably find Livewire handles a no longer authenticated user in the same way Laravel does. Can you test a standard Laravel controller and report back what it does? If it outputs a message then we will investigate why Livewire isn't.

Hope this helps!

**Comment by adeiming** [comment]
Oh sorry my laravel is v7.30.4 and my livewire is v2.3.16.

My application uses 2 routers, standard and livewire. if without the livewire it works fine, the 401 response code is unauthorized with a json response:
```
{message: "Unauthenticated."} 
```
But if using livewire, response code 302 but no data displayed like this: 

![Capture](https://user-images.githubusercontent.com/13284072/108018542-a3a01400-704a-11eb-97da-c3f97382f6b7.PNG)

and below the request leads to the login route.

![2](https://user-images.githubusercontent.com/13284072/108018681-fd084300-704a-11eb-86ad-7ca150e70488.PNG)

red: livewire 302 status.
blue: login route.

page response stuck like this:

![3](https://user-images.githubusercontent.com/13284072/108018910-92a3d280-704b-11eb-8398-85c795376571.PNG)

Thanks for response.

**Reply by mbagusrd** [reply]
baru nyadar kalo tombol hapusnya pake bahasa indo, :smile: 

salam kenal dari Gresik

**Comment by adeiming** [comment]
I finally solved the problem

in App\Http\Middleware\Authenticate.php
```
    protected function redirectTo($request)
    {
        if ($request->header('x-livewire')) return abort(401, 'Session has expired, please re-login');

        if (! $request->expectsJson()) {
            return route('login');
        }
    }
```

in app.blade.php
```
    window.livewire.onError(statusCode => {
      if (statusCode === 401) {
          window.location.href = '// login route';
          return false;
      }
    });
```

**Comment by ahmedsayedabdelsalam** [comment]
@calebporzio Isn't there any official solution for session expiration ?

now laravel autheticate middleware redirectTo method return 302 response to redirect to login but the page still the same.

**Comment by mbagusrd** [comment]
I think I have the same problem here, 

I have update password menu with livewire, when I successfully change my password then my session automatically expired and XHR request auto redirect to login route response. It makes livewire action paused without session expired confirmation dialog.

@calebporzio is there any solution for this condition?
